src/main/java/org/cloudfoundry/promregator/lite/discovery/RateLimiter.kt

```diff
@@ -0,0 +1,79 @@
+package org.cloudfoundry.promregator.lite.discovery
+
+import io.micrometer.core.instrument.MeterRegistry
+import kotlinx.coroutines.*
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.atomic.AtomicLong
+
+private val log = mu.KotlinLogging.logger {  }
+
+class RateLimiter(
```

Comment:
Since the only Guava class we use is for RateLimiting I implemented one specific to our needs. This uses the 'leaky bucket' type algorithm. 

Essentially the bucket has X tokens a minute, and it refills 1/60th of the tokens every second when it isn't full. This allows for initial bursts that might drain the bucket, then a trickle of requests as the bucket slowly refills.

It also provides 3 metrics: availableTokens, queueDepth, and totalWaitTime. Those are exposed elsewhere as metrics.